modules/db/src/main/java/org/mule/module/db/internal/domain/executor/StoredProcedureExecutor.java

```diff
@@ -40,7 +40,25 @@ protected Object doExecuteQuery(DbConnection connection, Statement statement, Qu
     @Override
     protected Object doExecuteQuery(DbConnection connection, Statement statement, Query query, AutoGeneratedKeyStrategy autoGeneratedKeyStrategy) throws SQLException
     {
-        autoGeneratedKeyStrategy.execute(statement, query.getQueryTemplate());
+        try
+        {
+            autoGeneratedKeyStrategy.execute(statement, query.getQueryTemplate());
```

Comment:
This try/catch must go inside the different AutoGeneratedKeyStrategy implementatations (maybe defining an abstract class to avoir repeating the code)